hibernate/hibernate-orm
# HHH-10999 Array datatypes module

What works:
- Using single-dimensional Object arrays in Entities (no primitive arrays).
- Saving arrays through entities through the `EntityManager` interface.
- Loading arrays into entities through the `EntityManager` interface.
- Tested under PostgreSQL.

What doesn't work:
- Table creation without setting `columnDefinition` on the `javax.persistence.Column` annotation.
- Passing null in the position of an array in manually written queries (bug is in Postgre server, not Hibernate).
- Using arrays of primitives, which are optional for JDBC drivers, even when arrays are supported.

TODO list without changes to core:
- Test more types
- Determine which types should be contributed, and which should be optional.
- Test on HyperSQL (only other database listed in the test suite that supports arrays).
- Annotate the tests to only use compatible dialects or features

TODO list with core changes:
- Add dialect feature checks for arrays to use for testing.
- Change StandardTableExporter to recognize and use arrays when desired.
- Use option to determine to use legacy model (default) or new array model.
